### PR TITLE
Onboarding tour QA updates

### PR DESCRIPTION
**Pull request by attfarhan**
Fixes https://github.com/sourcegraph/sourcegraph/issues/13396.

- Moves the search reference step to the search results page, as the last step.
- Adds a "Next" button to the structural search step:
Screen Shot 2020-08-28 at 3 07 49 PM
- Removes line break after the `return` key on the search button step.

### CODE REVIEW

File: web/src/integration/search-onboarding.test.ts [hunk 1]
```diff
@@ -73,13 +73,15 @@ describe('Search onboarding', () => {
...
             await driver.page.waitForSelector('.test-tour-step-4')
-            await driver.page.click('.test-search-help-dropdown-button-icon')
-            await driver.page.waitForSelector('.test-tour-step-5')
             await driver.page.click('.test-search-button')
             await driver.assertWindowLocation(
                 '/search?q=lang:typescript+try%7B:%5Bmy_match%5D%7D&patternType=structural&onboardingTour=true'
             )
+            await driver.page.waitForSelector('.test-tour-step-5')
+            await driver.page.waitForSelector('.tour-structural-next-button')
+            await driver.page.click('.tour-structural-next-button')
```
Comment (felixfbecker) [inline web/src/integration/search-onboarding.test.ts:82]: **nitpick**: CSS classes in tests should be stable `test-*` classes, non-prefixed classes should be for styling only

File: web/src/search/input/SearchOnboardingTour.ts [hunk 2]
```diff
@@ -215,11 +214,45 @@ export function createAddCodeStepWithLanguageExampleTooltip(
...
+export function createStructuralSearchTourTooltip(tour: Shepherd.Tour): HTMLElement {
+    const container = document.createElement('div')
+    const list = document.createElement('ul')
+    list.className = 'caret-list mb-0'
+    const listItem = document.createElement('li')
+    listItem.className = 'p-0 my-4'
+    list.append(listItem)
+    const exampleButton = document.createElement('a')
+    exampleButton.href = 'https://docs.sourcegraph.com/user/search/structural'
+    exampleButton.target = '_blank'
```
Comment (felixfbecker) [inline web/src/search/input/SearchOnboardingTour.ts:233]: **issue**: `target="_blank"` always needs `rel="noopener"`
```suggestion
    exampleButton.target = '_blank'
    exampleButton.rel = 'noopener'
```

File: web/src/search/input/SearchOnboardingTour.ts [hunk 3]
```diff
@@ -215,11 +214,45 @@ export function createAddCodeStepWithLanguageExampleTooltip(
...
         'Type the name of a function, variable or other code. Or try an example:',
         list
     )
 }
 
+/** Creates the tooltip for the structural search informational step. */
+export function createStructuralSearchTourTooltip(tour: Shepherd.Tour): HTMLElement {
+    const container = document.createElement('div')
+    const list = document.createElement('ul')
+    list.className = 'caret-list mb-0'
```
Comment (felixfbecker) [inline web/src/search/input/SearchOnboardingTour.ts:227]: **question**: Wasn't `caret-list` removed? cc @rrhyne

**Review by felixfbecker — COMMENTED** [review]
Deferring to @limitedmage for the project-specific review

File: web/src/search/input/SearchOnboardingTour.ts (hunk 3 above)
Comment (attfarhan) [reply web/src/search/input/SearchOnboardingTour.ts]: Just merged main to fix 👍

**Review by felixfbecker — APPROVED** [review]

**Comment by limitedmage** [comment]
**minor**: The triangle on the tip is (almost) invisible in the search results page (last step of the tour)
image

**Comment by limitedmage** [comment]
**minor:** In the structural search tip, the bullet point is not aligned with the text
image

**Comment by limitedmage** [comment]
There seems to be an odd scrolling issue when displaying the tips in the search results page. The whole page is scrollable with a dark block at the bottom while the tips are showing.

Screen Recording 2020-08-31 at 11 18 45 AM

File: web/src/search/input/SearchPageInput.tsx [hunk 4]
```diff
@@ -173,24 +174,11 @@ export const SearchPageInput: React.FunctionComponent<Props> = (props: Props) =>
...
-                    },
-                    advanceOn: { selector: '.search-help-dropdown-button', event: 'click' },
-                },
-                {
-                    id: 'final-step',
+                    id: 'submit-search',
                     text: generateStepTooltip(
                         tour,
-                        'Use <kbd>return</kbd><br/>or the search button to run your search',
+                        'Use <kbd>return</kbd> or the search button to run your search',
```
Comment (limitedmage) [inline web/src/search/input/SearchPageInput.tsx:180]: **minor:** On different platforms, this key is called differently (eg. 'enter' on Windows). Not sure if we have style guidelines for naming keys that are called differently on different platforms, but that may be something we want to consider. More of a copy/design question than anything.

Comment (attfarhan) [reply web/src/search/input/SearchPageInput.tsx:180]: cc @sourcegraph/design -- I know there's been discussion around a styleguide for copy

**Comment by attfarhan** [comment]
Thanks @limitedmage! I fixed two issues -- the one I couldn't quite get right is how to get the arrow to be more visible. @rrhyne please lmk if you have CSS knowhow for this.

**Comment by rrhyne** [comment]
> Thanks @limitedmage! I fixed two issues -- the one I couldn't quite get right is how to get the arrow to be more visible. @rrhyne please lmk if you have CSS knowhow for this.

Setting the entire background to $grey-19 seems seems to work. I couldn't try it live, but here's a screenshot from figma:

image

**Pull request merged**